dips/dip-161.md

```diff
@@ -0,0 +1,52 @@
+---
+dip: 161
+title: Optional Result Field in Off-Chain Protocol
+authors: Sunmi Lee (@sunmilee), David Wolinsky (@davidiw)
+status: Draft
+type: Informational
+created: 04/08/2020
+last_updated: 04/08/2021
+issue: https://github.com/diem/dip/issues/161
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# Overview
+
+This DIP describes an optional result field in Off-Chain command response, and is an extension of [DIP-1](https://github.com/diem/dip/blob/main/dips/dip-1.mdx).
+Both the Off-Chain Protocol and Travel Rule Exchange (or PaymentCommand) are defined in DIP-1. The Off-Chain Protocol was defined in such a way as to support the minimal requirements for these exchanges, specifically, in this case, asynchronous responses to requests. Many types of communication benefit from synchronous responses as it provides for both protocol simplicity and reduced latency. 
+
+In order to support synchronous commands, the `CommandResponseObject` can leverage an optional `result` field. The `result` field, if defined, must contain `__ObjectType` that uniquely defines other fields within the result.
+
+Here's an example from DIP-10:
+```
+{
+   "_ObjectType": "CommandResponseObject",
+    "status": "success",
+    "result": {
+	    "_ObjectType": "ReferenceIDCommandResponse",
+	    "receiver_address": "c5ab123458df0003415689adbb47326d",
+    },
+    "cid": "12ce83f6-6d18-0d6e-08b6-c00fdbbf085a",
+}
+```
+
+**CommandResponseObject:**
+
+| Field 	    | Type 	     | Required? 	| Description 	           |
+|-------	    |------	     |-----------	|-------------	           |
+| _ObjectType   | str        | Y | Fixed value: `CommandResponseObject`|
+| status       | str | Y | Either `success` or `failure`. |
+| result       | CommandResultObject() | Y | The Result obejct of response. |
+| cid           | str         | Y            | A unique identifier for the Command. Should be a UUID according to [RFC4122](https://tools.ietf.org/html/rfc4122) with "-"'s included. |
+
+**CommandResultObject:**
+
+***Type: ReferenceIDCommandResponse***
+
+| Field 	    | Type 	     | Required? 	| Description 	           |
+|-------	    |------	     |-----------	|-------------	           |
+| _ObjectType   | str        | Y | Fixed value: `ReferenceIDCommandResponse`|
+| recipient_address       | str | Y | Receiver's on-chain address |
```

Comment:
feels awkward to be defining ReferenceIDCommandResponse here, this should be more AbstractCommandResponse or something